pkg/cloudcore/apis/config/types.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import commonconfig "github.com/kubeedge/kubeedge/pkg/common/apis/config"
+
+type CloudCoreConfig struct {
+	Kube             *KubeConfig             `json:"kube,omitempty"`
+	EdgeController   *EdgeControllerConfig   `json:"edgeController,omitempty"`
+	DeviceController *DeviceControllerConfig `json:"deviceController,omitempty"`
+	Cloudhub         *CloudHubConfig         `json:"cloudHub,omitempty"`
+	Modules          *commonconfig.Modules   `json:"modules,omitempty"`
+}
+
+type EdgeControllerConfig struct {
+	NodeUpdateFrequency int32              `json:"nodeUpdateFrequency,omitempty"`
+	ControllerContext   *ControllerContext `json:"Context"`
+}
+
+type DeviceControllerConfig struct {
+	ControllerContext *ControllerContext `json:"Context"`
+}
+
+type CloudHubConfig struct {
+	// enable websocket protocol ,default true
+	EnableWebsocket bool `json:"enableWebsocket,omitempty"`
+	// open port for websocket server, default 10000
+	WebsocketPort int32 `json:"websocketPort,omitempty"`
+	// enable quic protocol, default false
+	EnableQuic bool `json:"enableQuic,omitempty"`
+	// open prot for quic server, default 10001
+	QuicPort int32 `json:"quicPort,omitempty"`
+	// the max incoming stream for quic server, default 10000
+	MaxIncomingStreams int32 `json:"maxIncomingStreams,omitempty"`
+	// enable unix domain socket protocol, default true
+	EnableUnixSocket bool `json:"enableUnixSocket,omitempty"`
+	// unix domain socket address, default unix:///var/lib/kubeedge/kubeedge.sock
+	UnixSocketAddress string `json:"unixSocketAddress,omitempty"`
+	//default 0.0.0.0
+	Address string `json:"address,omitempty"`
+	//default /etc/kubeedge/ca/rootCA.crt
+	TLSCaFile string `json:"tlsCaFile,omitempty"`
```

Comment:
```suggestion
	TLSCAFile string `json:"tlsCAFile,omitempty"`
```

Reply:
unversiond component api config is useless,  i has deleted it